src/arche/report.py

```diff
@@ -1,105 +1,51 @@
-        for level, rule_msgs in result.messages.items():
-            for rule_msg in rule_msgs:
-                cls.write_rule_outcome(rule_msg.summary, level)
-
-    @classmethod
-    def write_rule_outcome(
-        cls, outcome: Union[str, Outcome], level: Level = Level.INFO
-    ) -> None:
-        if isinstance(outcome, Outcome):
-            outcome = outcome.name
-        msg = outcome
-        if level == Level.ERROR:
-            cls.write_color_text(msg)
-        elif level == Level.WARNING:
-            cls.write_color_text(msg, color="#CCCC00")
-        elif outcome == Outcome.PASSED.name:
-            cls.write_color_text(msg, color="#0B6121")
+    def __call__(self, rule: Result = None, keys_limit: int = None) -> None:
+        if not rule:
+            template = self.env.get_template("full-report.html")
+            resultHTML = template.render(
+                rules=sorted(self.results.values(), key=lambda x: x.outcome.value),
+                pd=pd,
+                linkfy_callbacks=[callbacks.target_blank],
+                keys_limit=keys_limit,
+            )
         else:
-            cls.write(msg)
-
-    def write_details(self, short: bool = False, keys_limit: int = 10) -> None:
-        display_markdown("<h2>Details</h2>")
-        for result in self.results.values():
-            if result.detailed_messages_count:
-                display_markdown(
-                    f"{result.name} ({result.detailed_messages_count} message(s)):"
-                )
-                self.write_rule_details(result, short, keys_limit)
-                display_markdown("<br>")
-        display_markdown("<h2>Plots</h2>")
-        for result in self.results.values():
-            for f in result.figures:
-                f.show()
-
-    @classmethod
-    def write_rule_details(
-        cls, result: Result, short: bool = False, keys_limit: int = 10
-    ) -> None:
-        for rule_msgs in result.messages.values():
-            for rule_msg in rule_msgs:
-                if rule_msg.errors:
-                    cls.write_detailed_errors(rule_msg.errors, short, keys_limit)
-                if rule_msg.detailed:
-                    cls.write(rule_msg.detailed)
-
-    @classmethod
-    def write_detailed_errors(cls, errors: Dict, short: bool, keys_limit: int) -> None:
-        error_messages = sorted(errors.items(), key=lambda i: len(i[1]), reverse=True)
-
-        if short:
-            keys_limit = 5
-            error_messages = error_messages[:5]
-
-        for attribute, keys in error_messages:
-            if isinstance(keys, list):
-                keys = pd.Series(keys)
-            if isinstance(keys, set):
-                keys = pd.Series(list(keys))
-
-            sample = Report.sample_keys(keys, keys_limit)
-            display_markdown(f"{len(keys)} items affected - {attribute}: {sample}")
+            template = self.env.get_template("single-rule.html")
+            resultHTML = template.render(
+                rule=rule,
```

Comment:
Couldn't it be just a rule wrapped in a list or something like that?
So, there is only one template (multiple entries) and this is a special case (where you wrap a value in a list.

Reply:
But the information is different, for single rule we add `summary` which goes into a separate report for all rules.

Reply:
@ejulio I don't think so. The information is displayed in a different way by using the single rule(the plots are displayed directly on the rule and not on a separated "plot" tab as on the full report.